taichi/program/program.cpp

```diff
@@ -39,13 +39,6 @@
 
 namespace taichi {
 namespace lang {
-namespace {
-inline uint64 *allocate_result_buffer_default(Program *prog) {
```

Comment:
nit: Maybe keep this but pass in the `memory_pool` itself? Your call..

Reply:
Was planning to do this memory allocation in each arch's impl so separated them out. ;) Let's keep it for now, we might be able to unify them to a proper API (possibly in memory_pool) in the future.